src/ghost_cell.rs

```diff
@@ -558,4 +613,28 @@ pub fn cell_get_mut_borrows_cell_mutably() {}
 /// ```
 pub fn cell_from_mut_borrows_value_mutably() {}
 
+#[cfg(feature = "experimental-multiple-mutable-borrows")]
+///```rust
+///fn multiple_borrows_test() {
+///    use ghost_cell::{GhostToken, GhostCell};
+///    let n = 42;
+///    let value = GhostToken::new(|mut token| {
+///        let cell1 = GhostCell::new(42);
+///        let cell2 = GhostCell::new(47);
+///        let cell3 = GhostCell::new(7);
+///        let cell4 = GhostCell::new(9);
+///        let (reference1, reference2, reference3, reference4): (&mut i32, &mut i32, &mut i32, &mut i32)
+///            = GhostCell::borrow_mut_four(&cell1, &cell2, &cell3, &cell4, &mut token).unwrap();
+///        *reference1 = 33;
+///        *reference2 = 34;
+///        *reference3 = 35;
+///        *reference4 = 36;
+///    // here we stop mutating, so the token isn't mutably borrowed anymore, and we can read again
+///        (*cell1.borrow(&token), *cell2.borrow(&token), *cell3.borrow(&token))
+///    });
+///    assert_eq!((33, 34, 35), value);
+///}
+///```
+pub fn multiple_borrows_test() {}
```

Comment:
Any reason this is a doc-test?

The above are doc-tests because they fail at compilation, which can only work with doc-tests, however for tests supposed to run it would be better to use normal tests. Even panics can be caught with `#[should_panic]`. It's really only compilation failures that require doc-tests.